src/App.tsx

```diff
@@ -1,7 +1,39 @@
-import React from 'react';
+import { Component } from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+interface State {
+  pressedKey: string | null;
+}
+
+export class App extends Component<{}, State> {
+  state = {
+    pressedKey: null,
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', this.handleKeyUp);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.handleKeyUp);
+  }
```

Comment:
It is not a mistake, but try to keep one style of code... if you give a type for methods do this always or never)
I mean look at 12 and 16 strings.. : void {}

Reply:
agree, just remove :void, bc typescript already know about this type